sdk/Pulumi.Automation/WorkspaceStack.cs

```diff
@@ -332,6 +332,12 @@ public async Task<UpResult> UpAsync(
                 if (options.ExpectNoChanges is true)
                     args.Add("--expect-no-changes");
 
+                if (options.ShowReads.HasValue)
+                {
+                    args.Add("--show-reads");
+                    args.Add(options.ShowReads.Value ? "true" : "false");
```

Comment:
Needs to be just "--show-reads" or "--show-reads=true/false" can't be space separated.